File: elixir/word-count/word_count.exs
defmodule Words do
  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """
  @spec count(String.t) :: map
  def count(sentence) do
    sentence
      |> String.downcase
      |> String.replace(~r/[_!&@$%^,:]/, " ")
      |> String.split(" ", trim: true) 
      |> Enum.reduce(%{}, &calculate/2) 
  end

  defp calculate(elem, result) do 
    Map.update(result, elem, 1, &(&1 + 1))
  end
end

File: javascript/word-count/word-count.js
const Words = () => {
    clean = input => {
        return input.replace(/\r?\n|\r|\t/g, ' ')
    }
    return {
        count: (input) => {
            return clean(input).split(' ').reduce((prev, curr) => {
                let newObj = Object.create(null)
                let value = curr.trim().toLowerCase()
                if (value) newObj[value] = (prev[value]) ? prev[value] + 1 : 1
                return Object.assign(prev, newObj)
            }, Object.create(null));
        }
    }
}

module.exports = Words